docs/design/robots.md

```diff
@@ -0,0 +1,198 @@
+#Service Accounts
+
+## Motivation 
+
+Processes in Pods may need to call the Kubernetes API.  For example:
+  - scheduler
+  - replication controller
+  - minion controller
+  - a map-reduce type framework which has a controller that then tries to make a dynamically determined number of workers and watch them
+  - continuous build and push system
+  - monitoring system
+
+Distinguishing between humans and service accounts (a.k.a. robot accounts or roles) is a practice used by a number of cloud providers, such as:
+    - Google [service accounts](https://developers.google.com/accounts/docs/OAuth2ServiceAccount), which can be attached
+      to VM instances.
+    - Amazon Web Services [Roles](http://docs.aws.amazon.com/IAM/latest/UserGuide/Using_CreateRole_CLIAPI.html), which
+      can be attached to VM instances.
+    - quay.io [robot accounts](http://docs.quay.io/glossary/robot-accounts.html)
+
+The requirements for humans and automated processes are different:
+    - Humans need a wide range of capabilities to do their daily activities. Automated processes often have more narrowly-defined activities.
+    - Humans may better tolerate the exceptional conditions created by expiration of a token.  Remembering to handle
+      this in a program is more annoying.  So, either long-lasting credentials or automated rotation of credentials is
+      needed.
+    - A Human typically keeps credentials on a machine that is not part of the cluster and so not subject to automatic
+      management.  A VM with a role/service-account can have its credentials automatically managed.
+
+Therefore, Kubernetes should support both human and service accounts, with different ways of managing credentials and
+policies for each.
+
+## Design Overview
+
+A new REST object is added to the apiserver that represents a service account, called `serviceAccount`.  Another new
+REST object is added to the apiserver that represents a secret which an account needs to prove its identity to peers.
+This is called `secret`.  A `serviceAccount` has access to one or more `secret` objects.  The storage of secrets is separate to allow listing `/serviceAccounts` without seeing secrets.
+
+The `PodSpec` is extended to allow Pods to be started with an associated serviceAccount.  
```

Comment:
There may also be good reasons for isolating security contexts of pods in the same namespace.  If possible, suggest avoid hardcoding a requirement for all pods in namespace to share uid.

If you want to avoid uid reuse